_rules/css-restrict-orientation-b33eff.md

```diff
@@ -30,13 +30,13 @@ The rule applies to any element that is [visible](#visible) and has a CSS [trans
 - [matrix](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix)
 - [matrix3d](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix3d)
 
-that are applied conditionally on the [orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation) [media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features) with a value of `landsapce` or `portrait`.
+that are applied conditionally on the [orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation) [media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features) with a value of `landscape` or `portrait`.
 
 **Note:** These specific [transformation functions](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function) are of interest to this rule as they have the potential to affect the [rotation](https://drafts.csswg.org/css-transforms-2/#Rotate3dDefined) of a given element.
 
 ## Expectation
 
-The target element has a [current transformation matrix](https://drafts.csswg.org/css-transforms/#current-transformation-matrix) that does not restrict the element to either `landspace` or `portrait` orientation.
+The target element is neither rotated clockwise nor counter clockwise around the Z-axis at an angle corresponding to 90 degrees relative to the position of the element prior to transformation.
```

Comment:
We're getting there! 🙌 There are however still some issues we need to iron out though. Consider this bit of CSS for starters:

```css
body {
  transform: rotate(90deg);
}

@media (orientation: portrait) {
  body {
    transform: rotate(90deg);
  }
}
```

This would currently fail the rule but I'm having a hard time justifying why it should. The conditional rule makes no difference in practice; the `<body>` element is always rotated 90 degrees regardless of the orientation. I'm thinking we need to consider the rotation of an element at a given orientation relative to the rotation of that element at the opposite orientation. For the example above, the `<body>` element is effectively rotated 0 degrees in portrait orientation relative to its rotation in landscape rotation and vice versa.

One possible solution to this could be to use two expectations:

> ## Expectation 1
>
> In portrait orientation, the target element is neither rotated clockwise nor counter clockwise around the Z-axis at an angle corresponding to 90 degrees relative to the position of the element in landscape orientation.
>
> ## Expectation 2
>
> In landscape orientation, the target element is neither rotated clockwise nor counter clockwise around the Z-axis at an angle corresponding to 90 degrees relative to the position of the element in portrait orientation.

It does seem a little excessive but might be enough to get some ideas flowing.

Reply:
My suggestion above would also make it clear that testers actually need to look at both orientations to determine if there's an orientation lock or not, which makes perfect sense. Consider even a simple case such as this:

```css
@media (orientation: portrait) {
  body {
    transform: rotate(90deg);
  }
}
```

If this rule is only ever evaluated in landscape mode by a tester, the expectation would not fail.

Reply:
These are valid points & captures the idea of [essential](https://www.w3.org/WAI/WCAG21/Understanding/orientation.html#dfn-essential) very well.

I have made the changes to expectation & removed the assumption on `essential` as well as added a test case to showcase above case. Ta

Reply:
My points have nothing to do with whether the orientation lock is essential or not though; it is still assumed that it isn't.